src/Microsoft.AspNet.Razor.Runtime/DefaultTagHelperContent.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Razor.Runtime
+{
+    /// <summary>
+    /// Default implementation of <see cref="TagHelperContent"/>.
+    /// </summary>
+    public class DefaultTagHelperContent : TagHelperContent, ITextWriterCopyable
+    {
+        private readonly BufferEntryCollection _buffer;
+
+        /// <summary>
+        /// Instantiates a new instance of <see cref="DefaultTagHelperContent"/>.
+        /// </summary>
+        public DefaultTagHelperContent()
+        {
+            _buffer = new BufferEntryCollection();
+        }
+
+        /// <inheritdoc />
+        public override bool IsModified
+        {
+            get
+            {
+                return _buffer.IsModified;
+            }
+        }
+
+        /// <inheritdoc />
+        public override bool IsWhiteSpace
+        {
+            get
+            {
+                foreach (var value in _buffer)
+                {
+                    if (value.Trim().Length != 0)
+                    {
+                        return false;
+                    }
+                }
+
+                return true;
+            }
+        }
+
+        /// <inheritdoc />
+        public override bool IsEmpty
+        {
+            get
+            {
+                foreach (var value in _buffer)
+                {
+                    if (!string.IsNullOrEmpty(value))
+                    {
+                        return false;
+                    }
+                }
+
+                return true;
+            }
+        }
+
+        /// <inheritdoc />
+        public override void SetContent(string value)
+        {
+            _buffer.Clear();
+            Append(value);
+        }
+
+        /// <inheritdoc />
+        public override void Append(string value)
+        {
+            _buffer.Add(value);
+        }
+
+        /// <inheritdoc />
+        public override void Clear()
+        {
+            _buffer.Clear();
+        }
+
+        /// <inheritdoc />
+        public override string GetContent()
+        {
+            return string.Join(string.Empty, _buffer);
+        }
+
+        /// <inheritdoc />
+        public override void CopyTo([NotNull]TagHelperContent tagHelperContent)
```

Comment:
Hmmm, should our `CopyTo`s be clearing the content? Usually when a `CopyTo` is implemented it's given an offset as to where it should start copying, otherwise it overwrites.

/cc @dougbu @rynowak